pkg/packagekit/kpatch.jsx

```diff
@@ -0,0 +1,376 @@
+            if (this.state.justCurrent) {
+                install = new Promise((resolve, reject) => {
+                    cockpit.spawn(["dnf", "-y", "kpatch", "manual"], { superuser: "require", err: "message" })
+                            .then(() => {
+                                if (!this.state.patchUnavailable && !this.state.patchInstalled)
+                                    cockpit.spawn(["dnf", "-y", "install", this.state.patchName], { superuser: "require", err: "message" }).then(resolve)
+                                            .catch(reject);
+                                else
+                                    resolve();
+                            })
+                            .catch(reject);
+                });
+            } else {
+                install = cockpit.spawn(["dnf", "-y", "kpatch", "auto"], { superuser: "require", err: "message" });
+            }
+            install.then(() => {
+                this.kpatchService.enable().then(() => {
+                    this.kpatchService.start().then(() => {
+                        this.setState({ showModal: false, error: "" });
+                    });
+                });
+            })
+                    .catch(e => this.setState({ error: e.toString() }))
+                    .finally(() => {
+                        this.checkSetup();
+                        this.setState({ updating: false });
+                    });
+        } else {
+            cockpit.spawn(["dnf", "-y", "kpatch", "manual"], { superuser: "require", err: "message" }).then(() => {
+                this.kpatchService.disable().then(() => {
+                    this.kpatchService.stop().then(() => {
+                        this.setState({ showModal: false, error: "" });
+                    });
+                });
+            })
+                    .catch(e => this.setState({ error: e.toString() }))
+                    .finally(() => {
+                        this.checkSetup();
+                        this.setState({ updating: false });
+                    });
+        }
+    }
+
+    render() {
+        // Not yet recognized
+        if (this.state.loaded === false || this.state.patchName === null)
+            return null;
+
+        // Not supported on this system
+        if (this.state.unavailable.length > 0)
+            return null;
+
+        let state = _("Enabled");
+        let actionText = _("Edit");
+        let action = () => this.setState({ showModal: true });
+
+        if (this.state.missing.length > 0 && this.state.unavailable.length === 0) {
+            state = _("Not installed");
+            actionText = _("Install");
+            action = this.handleInstall;
+        } else if (!this.state.enabled) {
+            state = _("Disabled");
+            actionText = _("Enable");
+        }
+
+        let kernel_name = "";
+        if (this.state.kernelName)
+            kernel_name = " (" + this.state.kernelName + ")";
```

Comment:
General code style suggestion: Make this const and use ?: . Similar to `error` below.